File: app-face/src/main/java/com/element/facex/ResultFragment.java
package com.element.facex;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

public class ResultFragment extends DialogFragment {

    private String message;

    private int iconResId;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(DialogFragment.STYLE_NORMAL, 0);
        setCancelable(false);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_server_auth_result, container, false);

        TextView confirmText = rootView.findViewById(R.id.confirmText);
        ImageView confirmImage = rootView.findViewById(R.id.confirmImage);

        confirmText.setText(message);
        confirmImage.setImageResource(iconResId);

        Button retryButton = rootView.findViewById(R.id.retryButton);
        retryButton.setVisibility(getActivity() instanceof ServerAuthActivity ? View.VISIBLE : View.GONE);
        retryButton.setOnClickListener(view -> {
            if (getActivity() instanceof ServerAuthActivity) {
                ((ServerAuthActivity) getActivity()).recapture();
            }
        });

        Button exitButton = rootView.findViewById(R.id.exitButton);
        exitButton.setOnClickListener(view -> {
            if (getActivity() != null) {
                getActivity().finish();
            }
        });

        return rootView;
    }

    void setData(String message, int iconResId) {
        this.message = message;
        this.iconResId = iconResId;
    }
}

File: app-face/src/main/java/com/element/facex/ResultFragment.kt
package com.element.facex

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import com.element.camera.ElementFaceCaptureActivity

class ResultFragment : DialogFragment() {

    private var message: String? = null

    private var iconResId = 0

    companion object {
        fun show(activity: ElementFaceCaptureActivity, message: String, iconResId: Int) {
            val resultFragment = ResultFragment().apply {
                setData(message, iconResId)
            }

            activity.supportFragmentManager.beginTransaction()
                .add(resultFragment, null)
                .commitAllowingStateLoss()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(STYLE_NORMAL, 0)
        isCancelable = false
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        val rootView = inflater.inflate(R.layout.fragment_server_auth_result, container, false)
        val confirmText = rootView.findViewById<TextView>(R.id.confirmText)
        val confirmImage = rootView.findViewById<ImageView>(R.id.confirmImage)
        val retryButton = rootView.findViewById<Button>(R.id.retryButton)
        val exitButton = rootView.findViewById<Button>(R.id.exitButton)

        confirmText.text = message
        confirmImage.setImageResource(iconResId)

        retryButton.visibility = if (activity is ServerSideAuthActivity) View.VISIBLE else View.GONE
        retryButton.setOnClickListener {
            (activity as? ServerSideAuthActivity)?.relaunch()
        }

        exitButton.setOnClickListener {
            activity?.finish()
        }

        return rootView
    }

    fun setData(message: String?, iconResId: Int) {
        this.message = message
        this.iconResId = iconResId
    }
}